### PR TITLE
Allow public/discoverable resource to be part of collection resource

### PR DESCRIPTION
**Pull request by pkdash**
### Pull Request Checklist: 
- [x] Positive Test Case Written by Dev

- [x] Automated Testing

- [ ] Sufficient User and Developer Documentation

- [x] Passing Jenkins Build

- [ ] Peer Code review and approval

### Positive Test Case
1. Create a collection resource.
2. Create a composite resource for which the owner is different from the collection resource. Set the composite resource to 'public'.
3. Edit the collection resource to add the composite resource to that collection. Verify that you are able to add the public composite resource to the collection.
4. Create another composite resource for which the owner is different from the collection resource. Set the composite resource to 'discoverable'.
5. Edit the collection resource to add the composite resource that you created in step#4 to that collection. Verify that you are able to add the discoverable composite resource to the collection.

### CODE REVIEW

**Comment by hydrocheck** [comment]
| METRIC | VALUE |
|:-----:|:-----:|
| NEW SECURITY HOTSPOTS | 0 |
| NEW DUPLICATED LINES DENSITY | 0.0 |
| NEW CODE SMELLS | 1 |
| NEW BUGS | 0 |
| NEW VULNERABILITIES | 0 |

[https://sonarqube.cuahsi-workstation.com:9000/dashboard?id=hydroshare-3981](https://sonarqube.cuahsi-workstation.com:9000/dashboard?id=hydroshare-3981)

File: hs_collection_resource/views.py [hunk 1]
```diff
@@ -117,13 +117,14 @@ def update_collection(request, shortkey, *args, **kwargs):
...
                 # or is owned by current user
                 is_shareable = res_to_add.raccess.shareable
+                is_public = res_to_add.raccess.public
                 is_owner = res_to_add.raccess.owners.filter(pk=user.pk).exists()
-                if not is_shareable and not is_owner:
-                        raise Exception('Only resource owner can add a non-shareable '
-                                        'resource to a collection ')
+                if not is_shareable and not is_public and not is_owner:
+                    raise Exception('Only resource owner can add a non-shareable or private '
+                                    'resource to a collection ')
```
Comment (hyi) [inline hs_collection_resource/views.py:127]: @pkdash @alvacouch @sblack-usu Should we allow discoverable resources to be added into collection resources as well? I believe the code here only allows public and published resources, NOT the discoverable resources, to be added into collection resources.

Comment (pkdash) [reply hs_collection_resource/views.py:127]: @hyi I am not sure on this.

Comment (hyi) [reply hs_collection_resource/views.py:127]: @pkdash We discussed this on the coordination call and the team has decided that we should allow discoverable resources to be added in the collection in the same way as public resources.

Comment (pkdash) [reply hs_collection_resource/views.py:127]: @hyi Sounds good. Will implement that change soon.

Comment (pkdash) [reply hs_collection_resource/views.py:128]: @hyi I have now made code changes to allow discoverable to be part of a collection resource.

**Review by hyi — APPROVED** [review]

**Pull request merged**